commit da863481b7913c98f09490edb94771e9de83f5c1 (refs/changes/40/2440/1)
Author: Marsell Kukuljevic <marsell@joyent.com>
Date:   2017-08-22T23:27:25+12:00 (2 years, 2 months ago)
    
    PUBAPI-1427: There is no logging at all from the boostrap server

diff --git a/main.js b/main.js
index 9c2f3f4..2c61546 100644
--- a/main.js
+++ b/main.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2014, Joyent, Inc.
+ * Copyright (c) 2017, Joyent, Inc.
  */
 
 /*
@@ -24,6 +24,7 @@ var path = require('path');
 var bunyan = require('bunyan');
 var nopt = require('nopt');
 var restify = require('restify');
+var auditLogger = require('./lib/audit_logger');
 var RequestCaptureStream = restify.bunyan.RequestCaptureStream;
 
 var app = require('./lib').app;
@@ -162,12 +163,18 @@ function configure(file, options, log) {
 
 
 // Create a temporary server which simply returns 500 to all requests
-function createBootstrapServer(port, cb) {
-    var bootstrapServer = restify.createServer();
+function createBootstrapServer(port, log, cb) {
+    var bootstrapServer = restify.createServer({ log: log });
     bootstrapServer.use(restify.fullResponse());
 
+    bootstrapServer.on('after', auditLogger({
+        log: log.child({ component: 'bootstrap' })
+    }));
+
     function bootstrapHandler(req, res, next) {
-        res.send(new restify.InternalError('Failure to connect to Moray'));
+        var msg = 'Attempting to connect to internal services during bootstrap';
+        log.warn(msg);
+        res.send(new restify.InternalError(msg));
         next();
     }
 
@@ -206,7 +213,7 @@ function run() {
     // listening due to dependencies not being available (e.g. Moray is
     // offline, so plugin configs cannot be loaded)
 
-    createBootstrapServer(config.port, function (err, bootstrapServer) {
+    createBootstrapServer(config.port, LOG, function (err, bootstrapServer) {
         if (err) {
             throw err;
         }
